app/uk/gov/hmrc/gform/formtemplate/JsonSchemeValidator.scala

```diff
@@ -67,59 +43,193 @@ object JsonSchemeValidator {
+    .map(_._2)
+    .headOption
+
+  private def parseConditionalValidationErrors(
+    errors: NonEmptyList[ValidationError],
+    firstDependencyIndex: Int,
+    schema: Json,
+    json: Json
+  ): List[String] =
+    getErrorLocationsAndProperties(errors, firstDependencyIndex).distinct.map { case (location, property) =>
+      val errorLocation = tryConvertErrorLocationToId(json, location)
+
+      val maybeRequirements: Option[String] =
+        propertyRequirementsFromSchema(schema: Json, property: String) match {
+          case Left(_) => None
+          case Right(requiredPropertyAndPattern) =>
+            Some(requiredPropertyValuesFromPattern(requiredPropertyAndPattern))
+        }
+
+      maybeRequirements match {
+        case None => s"$errorLocation: Could not find validation in the schema for property: $property"
+        case Some(requirements: String) =>
+          s"Error at ID <$errorLocation>: Property $property can only be used with $requirements"
+      }
+    }
+
+  private def getErrorLocationsAndProperties(
+    errors: NonEmptyList[ValidationError],
+    firstDependencyIndex: Int
+  ): List[(String, String)] =
+    errors.toList.slice(firstDependencyIndex, errors.length).flatMap { error =>
+      val schemaErrorLocation = error.schemaLocation.getOrElse("")
+      schemaErrorLocation match {
+        case "" => None
+        case _  => maybeErrorLocationAndPropertyFromKeyword(error, schemaErrorLocation.split("/"))
+      }
+    }
+
+  private def maybeErrorLocationAndPropertyFromKeyword(
+    error: ValidationError,
+    splitErrorLocation: Array[String]
+  ): Option[(String, String)] =
+    error.keyword match {
+      case "pattern" =>
+        Some(
+          (
+            error.location.substring(0, error.location.lastIndexOf("/")),
+            splitErrorLocation(splitErrorLocation.length - 3)
+          )
+        )
+      case "required" =>
+        Some((error.location, splitErrorLocation.last))
+      case _ => None
+    }
+
+  private def tryConvertErrorLocationToId(json: Json, location: String): String = {
+    val errorLocationSections = location.split("/").toList.tail
+
+    getErrorLocationId(json, errorLocationSections) match {
+      // If cannot get ID of location, use original location message instead
+      case None => location
+
+      case Some(errorLocationId: Json) =>
+        errorLocationId.asString match {
+          case None                                => location
+          case Some(errorLocationIdString: String) => errorLocationIdString
+        }
+    }
+  }
+
+  @tailrec
+  private def getErrorLocationId(json: Json, remainingSections: List[String]): Option[Json] =
+    remainingSections.headOption match {
+      // No more sections to traverse, so get ID of current section to return
+      case None =>
+        json.hcursor.downField("id").as[Json] match {
+          case Left(_)                      => None
+          case Right(errorLocationId: Json) => Some(errorLocationId)
```

Comment:
Do not repeat types in pattern matching. This is for consistency, since we don't do this in other places. Here and elsewhere.

Feel free to put types to `val`s etc. if you want though.